packages/features/ee/teams/lib/payments.ts

```diff
@@ -136,37 +156,29 @@ export const purchaseTeamOrOrgSubscription = async (input: {
   });
   return { url: session.url };
 
-  /**
-   * Determines the priceId depending on if a custom price is required or not.
-   * If the organization has a custom price per seat, it will create a new price in stripe and return its ID.
-   */
-  async function getPriceId() {
-    const fixedPriceId = isOrg
-      ? process.env.STRIPE_ORG_MONTHLY_PRICE_ID
-      : process.env.STRIPE_TEAM_MONTHLY_PRICE_ID;
-
-    if (!fixedPriceId) {
-      throw new Error(
-        "You need to have STRIPE_ORG_MONTHLY_PRICE_ID and STRIPE_TEAM_MONTHLY_PRICE_ID env variables set"
-      );
-    }
-
-    log.debug("Getting price ID", safeStringify({ fixedPriceId, isOrg, teamId, pricePerSeat }));
-
-    if (!pricePerSeat) {
-      return fixedPriceId;
-    }
-
-    const priceObj = await stripe.prices.retrieve(fixedPriceId);
-    if (!priceObj) throw new Error(`No price found for ID ${fixedPriceId}`);
+  async function createPrice({
+    isOrg,
+    teamId,
+    pricePerSeat,
+    billingPeriod,
+    product,
+    currency,
+  }: {
+    isOrg: boolean;
+    teamId: number;
+    pricePerSeat: number;
+    billingPeriod?: BillingPeriod;
+    product: Stripe.Product | string;
+    currency: string;
+  }) {
     try {
       const customPriceObj = await stripe.prices.create({
         nickname: `Custom price for ${isOrg ? "Organization" : "Team"} ID: ${teamId}`,
         unit_amount: pricePerSeat * 100, // Stripe expects the amount in cents
         // Use the same currency as in the fixed price to avoid hardcoding it.
-        currency: priceObj.currency,
-        recurring: { interval: "month" }, // Define your subscription interval
-        product: typeof priceObj.product === "string" ? priceObj.product : priceObj.product.id,
+        currency: currency,
+        recurring: { interval: billingPeriod === "MONTHLY" ? "month" : "year" }, // Define your subscription interval
```

Comment:
We need to fix this condition.

If `billingPeriod` is undefined it fallbacks to Yearly. For existing teams that haven't upgraded yet, billingPeriod would be undefined causing there billing to become Yearly.